Snowflake.Data/Core/ArrowResultSet.cs

```diff
@@ -0,0 +1,168 @@
+        {
+            columnCount = responseData.rowType.Count;
+            try
+            {
+                using (var stream = new MemoryStream(Convert.FromBase64String(responseData.rowsetBase64)))
+                {
+                    using (var reader = new ArrowStreamReader(stream))
+                    {
+                        var recordBatch = reader.ReadNextRecordBatch();
+                        _currentChunkRowCount = recordBatch.Length;
+                        _currentChunk = new ArrowResultChunk(recordBatch);
+                    }
+                }
+
+                this.sfStatement = sfStatement;
+                UpdateSessionStatus(responseData);
+
+                if (responseData.chunks != null)
+                {
+                    _totalChunkCount = responseData.chunks.Count;
+                    
+                    // TODO in SNOW-893835 - support for multiple chunks
+                    throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+                }
+
+                responseData.rowSet = null;
+
+                sfResultSetMetaData = new SFResultSetMetaData(responseData);
+
+                isClosed = false;
+
+                queryId = responseData.queryId;
+            }
+            catch(System.Exception ex)
+            {
+                s_logger.Error("Result set error queryId="+responseData.queryId, ex);
+                throw;
+            }
+        }
+
+        internal override async Task<bool> NextAsync()
+        {
+            ThrowIfClosed();
+
+            _currentChunkRowIdx++;
+            if (_currentChunkRowIdx < _currentChunkRowCount)
+            {
+                return true;
+            }
+
+            if (_totalChunkCount > 0)
+            {
+                // TODO in SNOW-893835 - support for multiple chunks
+                throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+            }
+
+            return false;
+        }
+
+        internal override bool Next()
+        {
+            ThrowIfClosed();
+
+            _currentChunkRowIdx++;
+            if (_currentChunkRowIdx < _currentChunkRowCount)
+            {
+                return true;
+            }
+
+            if (_totalChunkCount > 0)
+            {
+                // TODO in SNOW-893835 - support for multiple chunks
+                throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+            }
+
+            return false;
+        }
+
+        internal override bool NextResult()
+        {
+            return false;
+        }
+
+        internal override async Task<bool> NextResultAsync(CancellationToken cancellationToken)
+        {
+            return await Task.FromResult(false);
+        }
+
+        internal override bool HasRows()
+        {
+            if (isClosed)
+            {
+                return false;
+            }
+
+            return _currentChunkRowCount > 0 || _totalChunkCount > 0;
+        }
+
+        /// <summary>
```

Comment:
Ok, if we do not return to the beginning maybe we should extend the test with two (more than one) nexts/rewinds?